### PR TITLE
Fix #2270

### LINKED ISSUE #2270: Server crash on register
<b>OS:</b>
Ubuntu 16.04

<b>Servatrice version:</b>
6962777

___
<br>

https://gist.github.com/skwerlman/eeaf79a9c89b7cc8c887c2aa126a920c

This seems to be caused by #2259

### PR DESCRIPTION
**Pull request by woogerboy21**
Fix #2270
Check userinfo first before trying to use it.

### CODE REVIEW

**Comment by Daenyth** [comment]
I'd really like to know why this happens

**Comment by Daenyth** [comment]
What's the series of events that triggers this?

**Comment by ctrlaltca** [comment]
You connect to the server and issue no command at all (or register and then just wait).
The server sends a ping, then checks for ping timeout.
The method where this is implemented dereferences the pointer userInfo without checking if it's not null.

**Comment by ctrlaltca** [comment]
> Shouldn't userInfo be not-null once you're connected?

Not the way it's currently implemented: `cmdLogin` is accepted only when userInfo is null and i-s filled using the user-provided data.

> Does the server ping before it checks auth?

The client pings the server every few seconds; the server only checks if the last ping has been received before then `max_player_inactivity_time` seconds ago and disconnects the client.
The server uses a single timer to check this for all connected clients, and the timer signal is connected to the method's slot as soon as the client object is created.

File: common/server_protocolhandler.cpp [hunk 1]
```diff
@@ -387,7 +387,7 @@ void Server_ProtocolHandler::pingClockTimeout()
     if (timeRunning - lastDataReceived > server->getMaxPlayerInactivityTime())
         prepareDestroy();
 
-    if (QString::fromStdString(userInfo->privlevel()).toLower() == "none") {
+    if (!userInfo || QString::fromStdString(userInfo->privlevel()).toLower() == "none") {
```
Comment (ZeldaZach) [inline common/server_protocolhandler.cpp]: I'm personally a fan of `if (userInfo && ...)` but this is fine.

Comment (ctrlaltca) [inline common/server_protocolhandler.cpp]: `if (userInfo && ...)` would skip the check for not-logged in users.

**Pull request merged**